pkg/cache/cache.go

```diff
@@ -65,8 +65,19 @@ type Response struct {
 	// Proxy responds with this version as an acknowledgement.
 	Version string
 
+	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
+	ResourceMarshaled bool
+
 	// Resources to be included in the response.
 	Resources []Resource
+
+	// Marshaled Resources to be included in the response.
+	MarshaledResources []MarshaledResource
+}
+
+type MarshaledResource struct {
```

Comment:
Perhaps we should put this in the resource file: https://github.com/envoyproxy/go-control-plane/blob/master/pkg/cache/resource.go

Reply:
I can see it in both ways, keeping MarshaledResource in the cache.go makes the file resource.go more isolated. But I'm open to it, and it depends on the maintainers' preference.